src/Polly.Core/Strategy/OutcomePredicate.cs

```diff
@@ -161,19 +161,25 @@ public OutcomePredicate<TArgs> ConfigurePredicates<TResult>(Action<OutcomePredic
     {
         Guard.NotNull(configure);
 
-        OutcomePredicate<TArgs, TResult>? predicate = null;
-
         if (!_predicates.ContainsKey(typeof(TResult)))
         {
-            predicate = new OutcomePredicate<TArgs, TResult>();
-            _predicates[typeof(TResult)] = (predicate, () => predicate.CreateHandler());
-        }
-        else
-        {
-            predicate = (OutcomePredicate<TArgs, TResult>)_predicates[typeof(TResult)].predicate;
+            SetPredicates(new OutcomePredicate<TArgs, TResult>());
         }
 
-        configure(predicate);
+        configure((OutcomePredicate<TArgs, TResult>)_predicates[typeof(TResult)].predicate);
+        return this;
+    }
+
+    /// <summary>
+    /// Sets the predicates for the specified result type.
+    /// </summary>
+    /// <typeparam name="TResult">The result type to add a predicate for.</typeparam>
+    /// <param name="predicate">The configured predicates.</param>
```

Comment:
Mix of plural (_"predicates"_) and singular (_"a predicate"_) here.

Reply:
Renamed.